Gems/EMotionFX/Code/MCore/Source/Algorithms.cpp

```diff
@@ -348,10 +348,11 @@ namespace MCore
             AZ::Vector3& result = results[i];
             result.Set(0.0f, 0.0f, 0.0f);
 
-            size_t left = AZStd::min(i, sampleNum);
-            size_t right = AZStd::min(data.size() - 1 - i, sampleNum);
-            result += AZStd::accumulate(data.begin() + i - left, data.begin() + i + right + 1, AZ::Vector3(0));
-            result /= (float)(left + right + 1);
+            const size_t left = AZStd::min(i, sampleNum);
+            const size_t right = AZStd::min(data.size() - 1 - i, sampleNum);
+            const size_t minSample = AZStd::min(left, right); // Make sure to take the same amount of samples from both size to calculate average.
+            result += AZStd::accumulate(data.begin() + i - minSample, data.begin() + i + minSample + 1, AZ::Vector3(0));
```

Comment:
nit: We should update this to `aznumeric_cast<float>` now and avoid an old C style cast

Reply:
suggestion: We could use `AZ::Vector3::CreateZero()` which is maybe a little more readable, again totally up to you, just a random thing I spotted

Reply:
you mean result = AZ::Vector3::CreateZero() instead of result.Set(0, 0, 0)? I think the latter should be more readable, no?

Reply:
Yeah that's what I was thinking but it's not a huge deal